### PR TITLE
Pilgrim tests

### PR DESCRIPTION
**Pull request by thePurpleMonkey**
Added pilgrim tests, fixed some uncovered bugs, and added some testing helper functions.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -86,4 +102,289 @@ describe("utilities", function() {
       assert(!utilities.inMovementRange(state, {x:3, y:4}), "True for left-up diag");
     });
   });
-});
\ No newline at end of file
+});
+
+describe("Pilgrim", function() {
```
Comment (JasonLarson44) [inline test.js:107]: Good work on getting all of these tests up

File: test.js [hunk 2]
```diff
@@ -3,7 +3,23 @@ const rewire = require('rewire');
 
 const bot = rewire('./test_compiled_bot.js');
 let utilities = bot.__get__("utilities");
-let SPECS = bot.__get__("SPECS")
+let pilgrim = bot.__get__("pilgrim");
+let SPECS = bot.__get__("SPECS");
+
+generate_grid = (x, y, val = false) => {
```
Comment (JasonLarson44) [inline test.js:9]: Is this useful for generating a map for testing?

File: bots/mainbot/pilgrim.js [hunk 3]
```diff
@@ -251,24 +248,24 @@ pilgrim.move = (self, target) => {
 	if (pilgrim.path.length === 0) {
 		if (utilities.inMovementRange(self, target)) {
 			// Target in movement range, so we can move one square towards target
-			let dx = target.x - self.x;
-			let dy = target.y - self.y;
-			let step = {x: self.x, y: self.y};
+			let dx = target.x - self.me.x;
+			let dy = target.y - self.me.y;
+			let step = {x: self.me.x, y: self.me.y};
```
Comment (JasonLarson44) [inline bots/mainbot/pilgrim.js:253]: Wait, was this the bug you were referring to?

**Review by JasonLarson44 — APPROVED** [review]

File: test.js (hunk 2 above)
Comment (thePurpleMonkey) [reply test.js:9]: That's the idea. It was getting pretty messy hardcoding all those 2d list of lists in each and every test.

File: bots/mainbot/pilgrim.js (hunk 3 above)
Comment (thePurpleMonkey) [reply bots/mainbot/pilgrim.js:253]: Yeah. This entire block of code was completely messed up.

File: test.js (hunk 2 above)
Comment (snuchhi) [reply test.js:9]: this is good .. helps to write other tests also.Thanks

**Review by snuchhi — APPROVED** [review]
Impressed that the test actually helped you find bug in pilgrim code.

**Pull request merged**